presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/confluent/ConfluentSchemaRegistryTableDescriptionSupplier.java

```diff
@@ -0,0 +1,350 @@
+        }
+    }
+
+    private String resolveSubject(String candidate)
+    {
+        String subject = subjectsSupplier.get().get(candidate);
+        checkState(subject != null, "Subject '%s' not found", candidate);
+        return subject;
+    }
+
+    private Map<String, String> getAllSubjects()
+    {
+        try {
+            return schemaRegistryClient.getAllSubjects().stream()
+                    .collect(toImmutableMap(subject -> subject.toLowerCase(ENGLISH), identity()));
+        }
+        catch (IOException | RestClientException e) {
+            throw new RuntimeException("Failed to retrieve subjects from schema registry", e);
+        }
+    }
+
+    // Refresh mapping of topic to subjects.
+    // *Note: that this mapping only supports subjects that use the SubjectNameStrategy of TopicName.
+    // If another SubjectNameStrategy is used then the key and value subject must be encoded
+    // into the table name as follows:
+    // <table name>[&key-subject=<key subject>][&value-subject=<value subject]
+    // ex. kafka.default."my-topic&key-subject=foo&value-subject=bar"
+    private Map<String, TopicAndSubjects> getTopicAndSubjects()
+    {
+        ImmutableSetMultimap.Builder<String, String> topicToSubjectsBuilder = ImmutableSetMultimap.builder();
+        for (String subject : subjectsSupplier.get().values()) {
+            if (isValidSubject(subject)) {
+                topicToSubjectsBuilder.put(extractTopicFromSubject(subject), subject);
+            }
+        }
+        ImmutableMap.Builder<String, TopicAndSubjects> topicSubjectsCacheBuilder = ImmutableMap.builder();
+        for (Map.Entry<String, Collection<String>> entry : topicToSubjectsBuilder.build().asMap().entrySet()) {
+            String topic = entry.getKey();
+            TopicAndSubjects topicAndSubjects = new TopicAndSubjects(
+                    topic,
+                    getKeySubjectFromTopic(topic, entry.getValue()),
+                    getValueSubjectFromTopic(topic, entry.getValue()));
+            topicSubjectsCacheBuilder.put(topicAndSubjects.getTableName(), topicAndSubjects);
+        }
+        return topicSubjectsCacheBuilder.build();
+    }
+
+    @Override
+    public Optional<KafkaTopicDescription> getTopicDescription(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        requireNonNull(schemaTableName, "schemaTableName is null");
+        TopicAndSubjects topicAndSubjects = parseTopicAndSubjects(schemaTableName);
+
+        String tableName = topicAndSubjects.getTableName();
+        Optional<TopicAndSubjects> topicAndSubjectsFromCache = Optional.ofNullable(topicAndSubjectsSupplier.get().get(tableName));
+
+        // Use the topic from cache, if present, in case the topic is mixed case
+        String topic = topicAndSubjectsFromCache.map(TopicAndSubjects::getTopic).orElse(topicAndSubjects.getTopic());
```

Comment:
I think user custom data is more important that cached. So let's fallback to cached data if it is missing in `schemaTableName`. What do you think?